lib/slime/tree.ex

```diff
@@ -26,7 +26,12 @@ defmodule Slime.Tree do
       |> Enum.map(&to_branch/1)
     filter          = make_child_filter(indentation)
     {children, rem} = Enum.split_while(t, filter)
-    children_tree   = existing ++ build_tree(children)
+    sep = if existing == [] && children != [] && Application.get_env(:slime, :keep_lines) do
```

Comment:
I think this would also be clearer as a variable. It's hard to read at the moment.